File: Proyecto de Prueva/backend/src/main/java/com/example/genapp/model/Calificacion.java
package com.example.genapp.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.experimental.SuperBuilder;
import java.util.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.*;

@Entity
@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@JsonIdentityInfo(
    generator = ObjectIdGenerators.PropertyGenerator.class,
    property = "id",
    scope = Calificacion.class
)
public class Calificacion {

        @Id
            @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;
        private BigDecimal notafinal;
        private String observacion;
        private String trimestre;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "inscripcion_id")
    @JsonIgnoreProperties(value = {"hibernateLazyInitializer", "handler"}, allowSetters = true)
    private Inscripcion inscripcion;



    public Boolean estaAprobado() {
    // TODO: implementar
    return false;
    }
    public void rectificarNota(BigDecimal nuevaNota, String motivo) {
    // TODO: implementar
    
    }
}

File: Proyecto de Prueva/frontend/lib/models/calificacion.dart
import 'inscripcion.dart';

  class Calificacion {
    final int id;
  final String notaFinal;
  final String observacion;
  final String trimestre;
  final String inscripcionId;
  final Inscripcion? inscripcion;

    Calificacion({
      
      required this.id, required this.notaFinal, required this.observacion, required this.trimestre, required this.inscripcionId, this.inscripcion,
    });

    factory Calificacion.fromJson(Map<String, dynamic> json) {
      return Calificacion(
        
        id: json['id'] is int ? json['id'] : int.tryParse(json['id']?.toString() ?? '0') ?? 0, notaFinal: json['notafinal']?.toString() ?? '', observacion: json['observacion']?.toString() ?? '', trimestre: json['trimestre']?.toString() ?? '', inscripcionId: json['inscripcionid'] != null ? json['inscripcionid'].toString() : (json['inscripcion'] is Map ? json['inscripcion']['id']?.toString() : json['inscripcion']?.toString()) ?? '', inscripcion: json['inscripcion'] is Map<String, dynamic> ? Inscripcion.fromJson(json['inscripcion']) : null,
      );
    }

    Map<String, dynamic> toJson() {
      return {
        
        'id': id, 'notafinal': notaFinal, 'observacion': observacion, 'trimestre': trimestre, 'inscripcionid': inscripcionId,
      };
    }

    @override
    String toString() {
      return 'notaFinal: ${notaFinal}' + ", " + 'observacion: ${observacion}';
    }
  }
  